internal/connector/internal/api/dbapi/processor.go

```diff
@@ -0,0 +1,117 @@
+package dbapi
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/db"
+)
+
+type ProcessorDesiredState string
+type ProcessorStatusPhase string
+
+const (
+	ProcessorReady   ProcessorDesiredState = "ready"
+	ProcessorStopped ProcessorDesiredState = "stopped"
+	ProcessorDeleted ProcessorDesiredState = "deleted"
+
+	ProcessorStatusPhasePreparing      ProcessorStatusPhase = "preparing"      // set by kas-fleet-manager - user request
+	ProcessorStatusPhasePrepared       ProcessorStatusPhase = "prepared"       // set by kas-fleet-manager - worker
+	ProcessorStatusPhaseUpdating       ProcessorStatusPhase = "updating"       // set by kas-fleet-manager - user request
+	ProcessorStatusPhaseStopped        ProcessorStatusPhase = "stopped"        // set by kas-fleet-manager - user request
+	ProcessorStatusPhaseProvisioning   ProcessorStatusPhase = "provisioning"   // set by kas-agent
+	ProcessorStatusPhaseReady          ProcessorStatusPhase = "ready"          // set by the agent
+	ProcessorStatusPhaseFailed         ProcessorStatusPhase = "failed"         // set by the agent
+	ProcessorStatusPhaseDeprovisioning ProcessorStatusPhase = "deprovisioning" // set by kas-agent
+	ProcessorStatusPhaseDeleting       ProcessorStatusPhase = "deleting"       // set by the kas-fleet-manager - user request
+	ProcessorStatusPhaseDeleted        ProcessorStatusPhase = "deleted"        // set by the agent
+)
+
+var ValidProcessorDesiredStates = []string{
+	string(ProcessorReady),
+	string(ProcessorStopped),
+	string(ProcessorDeleted),
+}
+
+var AgentProcessorStatusPhase = []string{
+	string(ProcessorStatusPhaseProvisioning),
+	string(ProcessorStatusPhaseDeprovisioning),
+	string(ProcessorStatusPhaseStopped),
+	string(ProcessorStatusPhaseReady),
+	string(ProcessorStatusPhaseFailed),
+	string(ProcessorStatusPhaseDeleted),
+}
+
+type Processor struct {
+	db.Model
+	Name            string
+	NamespaceId     string
+	ProcessorTypeId string
+	Channel         string
+	DesiredState    ProcessorDesiredState
+	Owner           string
+	CloudProvider   string
```

Comment:
I'm not sure I understand how `CloudProvider`, `Region` and `MultiAZ` work for Processors 🤔  The `namespaceId` will reference a Connector Cluster that is pre-deployed by the user, so at least the `CloudProvider` and `Region` are already set and essentially inherited from the underlying OSD on which the namespace is "hosted".

`MultiAZ` will again depend on the capabilities of the underlying OSD hosting the namespace and FWICS, we don't actually have that information about the Connector Cluster (I could be completely missing something as I'm just getting used to Go and this codebase).

Reply:
I can remove them. I will wait for all feedback to be assembled before making changes.

Reply:
I agree that `CloudProvier`, `Region` and `MultiAZ` are tied to the namespace/OSD so we can remove them from here

Reply:
Removed.